### PR TITLE
Bug fix and metadata config extension.

### PR DESCRIPTION
**Pull request by caiush**

### CODE REVIEW

File: cookbooks/bcpc/templates/default/nova.conf.erb [hunk 1]
```diff
@@ -38,7 +38,7 @@ osapi_compute_extension=nova.api.openstack.compute.contrib.standard_extensions
 osapi_compute_extension=cobalt.nova.osapi.cobalt_extension.Cobalt_extension
 <% end %>
 <% if not node["bcpc"]["vendordata_driver"].nil? %>
-vendordata_driver = node["bcpc"]["vendordata_driver"].
+vendordata_driver = <%=node["bcpc"]["vendordata_driver"] %>
```
Comment (acaiafa) [inline cookbooks/bcpc/templates/default/nova.conf.erb]: you want to add a space here 
```ruby
<%= node["bcpc"]["vendordata_driver"] %>
```

File: cookbooks/bcpc/templates/default/nova.conf.erb [hunk 2]
```diff
@@ -38,7 +38,7 @@ osapi_compute_extension=nova.api.openstack.compute.contrib.standard_extensions
 osapi_compute_extension=cobalt.nova.osapi.cobalt_extension.Cobalt_extension
 <% end %>
 <% if not node["bcpc"]["vendordata_driver"].nil? %>
```
Comment (acaiafa) [inline cookbooks/bcpc/templates/default/nova.conf.erb]: could make this 
```ruby
 <% unless node["bcpc"]["vendordata_driver"].nil? %>
```

File: cookbooks/bcpc/templates/default/nova.conf.erb (hunk 1 above)
Comment (caiush) [inline cookbooks/bcpc/templates/default/nova.conf.erb]: Why, its not the convention we have used everywhere else. If we want to switch to that (perhaps a good thing), we should do that in a separate PR, i.e. change the spacing convention.

Comment (acaiafa) [inline cookbooks/bcpc/templates/default/nova.conf.erb]: ah ok just saw it and figured to point out a common pattern that is widely used. I noticed the rest of the templates are like this so my apologies.

Comment (caiush) [inline cookbooks/bcpc/templates/default/nova.conf.erb]: There is a lot of, cough, "cruft" in the recipes that need to be cleaned up and made more standard. We can at that to the list.

File: cookbooks/bcpc/templates/default/nova.conf.erb [hunk 3]
```diff
@@ -182,3 +182,12 @@ inject_partition=-2
 live_migration_flag="VIR_MIGRATE_UNDEFINE_SOURCE,VIR_MIGRATE_PEER2PEER,VIR_MIGRATE_LIVE,VIR_MIGRATE_PERSIST_DEST,VIR_MIGRATE_TUNNELLED"
 virt_type=<%=node['bcpc']['virt_type']%>
 use_virtio_for_bridges=True
+
+
+<% unless node["bcpc"]["vendordata_config"].nil? %>
+[metadata]
```
Comment (erhudy) [inline cookbooks/bcpc/templates/default/nova.conf.erb]: If `['bcpc']['vendordata_config']` is present but empty, there will be a `[metadata]` section without anything in it. Will that cause problems?

Comment (caiush) [inline cookbooks/bcpc/templates/default/nova.conf.erb]: Provides a way to add in custom metadata driver config into the BCPC cookbooks. Should be a no-op for the laptop env.

**Comment by erhudy** [comment]
What's the right way to test that this fixed something in terms of functionality? Obviously looking at the template output to verify it was actually replaced with a thing is something, but what should be tested in terms of the functionality of the metadata service?
